src/main/java/com/diffblue/deeptestutils/CompareWithFieldList.java

```diff
@@ -0,0 +1,226 @@
+package com.diffblue.deeptestutils;
+
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * <code>CompareWithFieldList</code> does a deep compare on objects and their
+ * members.
+ *
+ * @author <a href="http://diffblue.com">Diffblue</a>
+ */
+public final class CompareWithFieldList {
+
+  /**
+   * Private constructor to prevent instantiation of the class with static
+   * methods.
+   */
+  private CompareWithFieldList() {
+  }
+
+  /**
+   * Output stream for debugging.
+   */
+  private static PrintStream debugOut;
+
+  static {
+    String outPath = System.getenv("CWFL_OUTPUT_FILE");
+    if (outPath != null) {
+      try {
+        debugOut = new PrintStream(outPath);
+      } catch (FileNotFoundException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      debugOut = null;
+    }
+  }
+
+  /**
+   * Set of primitive classes.
+   */
+  private static HashSet<Class> primitives;
+
+  static {
+    primitives = new HashSet<Class>();
+    primitives.add(Boolean.class);
+    primitives.add(Character.class);
+    primitives.add(Byte.class);
+    primitives.add(Short.class);
+    primitives.add(Integer.class);
+    primitives.add(Long.class);
+    primitives.add(Float.class);
+    primitives.add(Double.class);
+  }
+
+  /**
+   * Returns a named field of a class.
+   *
+   * @param cls the <code>Class</code> to look at
+   * @param name the name of the field as <code>String</code> value
+   * @return the <code>Field</code> value
+   */
+  private static Field getField(final Class cls, final String name) {
+    Class currentClass = cls;
+    while (cls != null) {
+      try {
+        return currentClass.getDeclaredField(name);
+      } catch (NoSuchFieldException e) {
+        currentClass = currentClass.getSuperclass();
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Throws an exception with detailed error message.
+   *
+   * @param actual the original <code>Object</code>
+   * @param expected the <code>Object</code> compared to that differs
+   * @param prefix the difference prefix to get the error message as
+   *   <code>String</code>
+   */
+  private static void fail(
```

Comment:
Suggested tidy up, not important to do right now - Would it be better to refactor the fail method into a constructor of the UnexpectedValueException class?